File: app/src/main/java/com/android/letsgo/network/response/GeocoderResponse.java
package com.android.letsgo.network.response;

import com.google.gson.annotations.SerializedName;

public class GeocoderResponse {
    @SerializedName("response") ResponseData response;

    public ResponseData getResponse() {
        return response;
    }

    public class ResponseData {
        @SerializedName("GeoObjectCollection") GeoObjectCollection geoObjectCollection;

        public GeoObjectCollection getGeoObjectCollection() {
            return geoObjectCollection;
        }
    }
    public class GeoObjectCollection {
        @SerializedName("featureMember") FeatureMember featureMember;

        public FeatureMember getFeatureMember() {
            return featureMember;
        }
    }
    public class FeatureMember {
        @SerializedName("GeoObject") GeoObject geoObject;
        public String getFormatedAdress(){
            if(geoObject!=null && geoObject.metaDataProperty!=null &&
            geoObject.metaDataProperty.geocoderMetaData!=null &&
            geoObject.metaDataProperty.geocoderMetaData.address !=null){
                return geoObject.metaDataProperty.geocoderMetaData.address.formatted;
            } else return null;
        }

        public GeoObject getGeoObject() {
            return geoObject;
        }
    }

    public class GeoObject {
        @SerializedName("metaDataProperty")  MetaDataProperty metaDataProperty;

        public MetaDataProperty getMetaDataProperty() {
            return metaDataProperty;
        }
    }

    public class MetaDataProperty {
        @SerializedName("GeocoderMetaData") GeocoderMetaData geocoderMetaData;

        public GeocoderMetaData getGeocoderMetaData() {
            return geocoderMetaData;
        }
    }

    public class GeocoderMetaData {
        @SerializedName("Address") Address address;

        public Address getAddress() {
            return address;
        }
    }

    public class Address {
        @SerializedName("formatted") String formatted;

        public String getFormatted() {
            return formatted;
        }
    }
}

File: app/src/main/java/com/android/letsgo/network/response/GeocoderResponse.kt
package com.android.letsgo.network.response

import com.google.gson.annotations.SerializedName

class GeocoderResponse {
    @SerializedName("response")
    var response: ResponseData? = null

    fun getResponse(): ResponseData? {
        return response
    }

    class ResponseData {
        @SerializedName("GeoObjectCollection")
        var geoObjectCollection: GeoObjectCollection? = null
    }

    class GeoObjectCollection {
        @SerializedName("featureMember")
        var featureMember: FeatureMember? = null
    }

    class FeatureMember {
        @SerializedName("GeoObject")
        var geoObject: GeoObject? = null
        val formatedAdress: String?
            get() = if (geoObject != null && geoObject?.metaDataProperty != null
                && geoObject!!.metaDataProperty!!.geocoderMetaData != null
                && geoObject!!.metaDataProperty!!.geocoderMetaData!!.address != null) {
                geoObject!!.metaDataProperty!!.geocoderMetaData!!.address!!.formatted
            } else null
    }

    class GeoObject {
        @SerializedName("metaDataProperty")
        var metaDataProperty: MetaDataProperty? = null
    }

    class MetaDataProperty {
        @SerializedName("GeocoderMetaData")
        var geocoderMetaData: GeocoderMetaData? = null
    }

    class GeocoderMetaData {
        @SerializedName("Address")
        var address: Address? = null
    }

    class Address {
        @SerializedName("formatted")
        var formatted: String? = null
    }
}